tietokanta/src/main/resources/db/migration/V1_839__.sql

```diff
@@ -0,0 +1,127 @@
+    RETURNS VOID AS
+$$
+BEGIN
+    RAISE NOTICE 'Siirretään toteumat aikaväliltä % - %', alkupvm, loppupvm;
+    WITH x AS (
+        DELETE FROM toteuma_010101_191001 WHERE alkanut BETWEEN alkupvm AND loppupvm returning *
+    )
+    INSERT INTO toteuma SELECT * FROM x;
+END
+$$
+    LANGUAGE plpgsql;
+
+-- Siirretään toteumat vanhasta isosta partitiosta uuteen 9kk pitkään
+SELECT * FROM siirra_aikavalin_toteumat( '2019-10-01'::DATE, '2020-07-01'::DATE);
+
+-- Päivitetään check constraint vanhalle partitiolle
+ALTER TABLE toteuma_010101_191001
+    ADD CONSTRAINT toteuma_010101_191001_alkanut_check
+        CHECK (alkanut >= '0001-01-01' AND alkanut < '2019-10-01');
+
+-- Muokkaa toteuma taulun poistettu kenttä pelkästään boolean arvoksi
+-- Ensin muutetaan null arvot falseksi
+UPDATE toteuma set poistettu = false WHERE poistettu is null;
+UPDATE toteuma_010101_191001 set poistettu = false WHERE poistettu is null;
+UPDATE toteuma_191001_200701 set poistettu = false WHERE poistettu is null;
+UPDATE toteuma_200701_210101 set poistettu = false WHERE poistettu is null;
+UPDATE toteuma_210101_210701 set poistettu = false WHERE poistettu is null;
+UPDATE toteuma_210701_220101 set poistettu = false WHERE poistettu is null;
+UPDATE toteuma_220101_220701 set poistettu = false WHERE poistettu is null;
+UPDATE toteuma_220701_230101 set poistettu = false WHERE poistettu is null;
+UPDATE toteuma_230101_230701 set poistettu = false WHERE poistettu is null;
+UPDATE toteuma_230701_240101 set poistettu = false WHERE poistettu is null;
+UPDATE toteuma_240101_240701 set poistettu = false WHERE poistettu is null;
+UPDATE toteuma_240701_250101 set poistettu = false WHERE poistettu is null;
+UPDATE toteuma_250101_991231 set poistettu = false WHERE poistettu is null;
+-- Sitten muokataan itse kenttä, jotta sinne ei voi null arvoja enää lisätä
+ALTER TABLE toteuma ALTER COLUMN poistettu SET NOT NULL;
+ALTER TABLE toteuma_010101_191001 ALTER COLUMN poistettu SET NOT NULL;
+ALTER TABLE toteuma_191001_200701 ALTER COLUMN poistettu SET NOT NULL;
+ALTER TABLE toteuma_200701_210101 ALTER COLUMN poistettu SET NOT NULL;
+ALTER TABLE toteuma_210101_210701 ALTER COLUMN poistettu SET NOT NULL;
+ALTER TABLE toteuma_210701_220101 ALTER COLUMN poistettu SET NOT NULL;
+ALTER TABLE toteuma_220101_220701 ALTER COLUMN poistettu SET NOT NULL;
+ALTER TABLE toteuma_220701_230101 ALTER COLUMN poistettu SET NOT NULL;
+ALTER TABLE toteuma_230101_230701 ALTER COLUMN poistettu SET NOT NULL;
+ALTER TABLE toteuma_230701_240101 ALTER COLUMN poistettu SET NOT NULL;
+ALTER TABLE toteuma_240101_240701 ALTER COLUMN poistettu SET NOT NULL;
+ALTER TABLE toteuma_240701_250101 ALTER COLUMN poistettu SET NOT NULL;
+ALTER TABLE toteuma_250101_991231 ALTER COLUMN poistettu SET NOT NULL;
+-- set default false
+ALTER TABLE toteuma ALTER COLUMN poistettu SET DEFAULT FALSE;
+ALTER TABLE toteuma_010101_191001 ALTER COLUMN poistettu SET DEFAULT FALSE;
```

Comment:
meniskö NOT NULL ja DEFAULT FALSE samalla komennolla?

Reply:
Yritin, en osannut laittaa. (normaalisti kun lisätään uutta kolumnia create tablessa, niin onnistuu, mutta en tiedä miksi se aina valitti, kun alterissa koitin)